<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SettingsDatabase</class>
 <widget class="QWidget" name="SettingsDatabase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>558</width>
    <height>356</height>
   </rect>
  </property>
  <layout class="QFormLayout" name="formLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item row="1" column="0" colspan="2">
    <widget class="QCheckBox" name="m_checkUseTransactions">
     <property name="toolTip">
      <string>Note that turning this option ON will make saving of new messages FASTER, but it might rarely cause some issues with messages saving.</string>
     </property>
     <property name="text">
      <string>Use DB transactions when storing downloaded messages</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="QLabel" name="m_lblDataStorageWarning">
     <property name="styleSheet">
      <string notr="true">QLabel {
	margin-top: 12px;
}</string>
     </property>
     <property name="text">
      <string>WARNING: Note that switching to another data storage type will NOT copy existing your data from currently active data storage to newly selected one.</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="m_lblDatabaseDriver">
     <property name="text">
      <string>Database driver</string>
     </property>
     <property name="buddy">
      <cstring>m_cmbDatabaseDriver</cstring>
     </property>
    </widget>
   </item>
   <item row="4" column="1">
    <widget class="QComboBox" name="m_cmbDatabaseDriver"/>
   </item>
   <item row="5" column="0" colspan="2">
    <widget class="QStackedWidget" name="m_stackedDatabaseDriver">
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="m_pageSqlite">
      <layout class="QFormLayout" name="formLayout_15">
       <property name="fieldGrowthPolicy">
        <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item row="0" column="0" colspan="2">
        <widget class="QCheckBox" name="m_checkSqliteUseInMemoryDatabase">
         <property name="text">
          <string>Use in-memory database as the working database</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QLabel" name="m_lblSqliteInMemoryWarnings">
         <property name="text">
          <string>Usage of in-memory working database has several advantages and pitfalls. Make sure that you are familiar with these before you turn this feature on. Advantages:
&lt;ul&gt;
&lt;li&gt;higher speed for feed/message manipulations (especially with thousands of messages displayed),&lt;/li&gt;
&lt;li&gt;whole database stored in RAM, thus your hard drive can rest more.&lt;/li&gt;
&lt;/ul&gt;
Disadvantages:
&lt;ul&gt;
&lt;li&gt;if application crashes, your changes from last session are lost,&lt;/li&gt;
&lt;li&gt;application startup and shutdown can take little longer (max. 2 seconds).&lt;/li&gt;
&lt;/ul&gt;
Authors of this application are NOT responsible for lost data.</string>
         </property>
         <property name="textFormat">
          <enum>Qt::RichText</enum>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
         <property name="indent">
          <number>20</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="m_pageMysql">
      <layout class="QFormLayout" name="formLayout_16">
       <property name="fieldGrowthPolicy">
        <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item row="0" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="LineEditWithStatus" name="m_txtMysqlHostname" native="true"/>
         </item>
         <item>
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Port</string>
           </property>
           <property name="buddy">
            <cstring>m_spinMysqlPort</cstring>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="m_spinMysqlPort">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>0</height>
            </size>
           </property>
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>65536</number>
           </property>
           <property name="value">
            <number>3306</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="1">
        <widget class="LineEditWithStatus" name="m_txtMysqlDatabase" native="true"/>
       </item>
       <item row="2" column="1">
        <widget class="LineEditWithStatus" name="m_txtMysqlUsername" native="true"/>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Password</string>
         </property>
         <property name="buddy">
          <cstring>m_txtMysqlPassword</cstring>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="LineEditWithStatus" name="m_txtMysqlPassword" native="true"/>
       </item>
       <item row="4" column="1">
        <widget class="QCheckBox" name="m_checkMysqlShowPassword">
         <property name="text">
          <string>&amp;Show password</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_11">
         <item>
          <widget class="QPushButton" name="m_btnMysqlTestSetup">
           <property name="text">
            <string>Test setup</string>
           </property>
           <property name="flat">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="LabelWithStatus" name="m_lblMysqlTestResult" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="layoutDirection">
            <enum>Qt::RightToLeft</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="6" column="0" colspan="2">
        <widget class="QLabel" name="m_lblMysqlInfo">
         <property name="text">
          <string>Note that speed of used MySQL server and latency of used connection medium HEAVILY influences the final performance of this application. Using slow database connections leads to bad performance when browsing feeds or messages.</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Username</string>
         </property>
         <property name="buddy">
          <cstring>m_txtMysqlUsername</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>Working database</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Hostname</string>
         </property>
         <property name="buddy">
          <cstring>m_txtMysqlHostname</cstring>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
  <zorder>m_lblDatabaseDriver</zorder>
  <zorder>m_cmbDatabaseDriver</zorder>
  <zorder>m_stackedDatabaseDriver</zorder>
  <zorder>m_checkUseTransactions</zorder>
  <zorder>m_lblDataStorageWarning</zorder>
  <zorder>label_2</zorder>
  <zorder>label_11</zorder>
  <zorder>label_4</zorder>
  <zorder>label_5</zorder>
  <zorder>m_lblMysqlInfo</zorder>
 </widget>
 <customwidgets>
  <customwidget>
   <class>LabelWithStatus</class>
   <extends>QWidget</extends>
   <header>labelwithstatus.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>LineEditWithStatus</class>
   <extends>QWidget</extends>
   <header>lineeditwithstatus.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
